signer/src/bitcoin/utxo.rs

```diff
@@ -343,29 +343,15 @@ impl DepositRequest {
         let txid = bitcoin::Txid::from_byte_array(
             request.txid.try_into().map_err(|_| Error::TypeConversion)?,
         );
-
-        let vout = request
-            .output_index
-            .try_into()
-            .map_err(|_| Error::TypeConversion)?;
-
-        let max_fee = request
-            .max_fee
-            .try_into()
-            .map_err(|_| Error::TypeConversion)?;
+        let vout = request.output_index;
 
         let signer_bitmap = BitArray::ZERO; // TODO(326): Populate
 
-        let amount = request
-            .amount
-            .try_into()
-            .map_err(|_| Error::TypeConversion)?;
-
         Ok(Self {
             outpoint: OutPoint { txid, vout },
-            max_fee,
+            max_fee: request.max_fee,
             signer_bitmap,
-            amount,
+            amount: request.amount,
             deposit_script: ScriptBuf::from_bytes(request.spend_script),
             reclaim_script: ScriptBuf::from_bytes(request.reclaim_script),
             signers_public_key,
```

Comment:
I find the approach that does not do the decomposition a little easier to read. That's because I only need to look at the lines in return type to see what values were set to any particular field (assuming request is immutable). Under the decomposition approach, there might be some intermediate change to a variable that I need to look out for. Also, the non-decomposition approach is fewer lines of code and doesn't seem much more complex. But the destructuring done in this example is easy enough to read so I wouldn't sweat it if I were reviewing.

Anyway, I think that's the general "principle" for me when I'm writing code. It's not set in stone for me, so I'm also curious about how you (and others) think about this.

Reply:
Whereever possible I see `amount: request.amount` and want to see it be just `amount`. That said, I actually do agree with your position, I was just curious what you thought.